### DISCUSSION TITLE
extra div being rendered on Next JS 13

### DISCUSSION THREAD

**Discussion by gmwill934** [vercel/next.js#41912] [Help] [answered]
I currently am testing the new features for Next JS 13, specifically the `app` folder ❤️ .

I have two files on that directory

`layout.tsx`

```
import { ReactNode } from 'react';
import './globals.css';

type RootLayoutProps = { children: ReactNode };

export default function RootLayout({ children }: RootLayoutProps) {
  return (
    <html>
      <head></head>
      <body>
        <section>{children}</section>
      </body>
    </html>
  );
}

```

`page.tsx`

```
export default function Home() {
  return <main>Home Page</main>;
}

```

If I navigate to the root of my application, I expect to have the following HTML structure inside my `body` element.

```
<body>
   <section>
      <main>
           <!-- Content here -->
      </main>
   <section>
</body>
```

Instead I get an extra `div` that's wrapping my `main` element

```
<body>
   <section>
      <div>
          <main>
               <!-- Content here -->
          </main>
      </div>
   <section>
</body>
```

Am I missing something?

**Comment by icyJoseph** [comment, marked as answer]
Yes, this has been observed and addressed here as well: https://github.com/vercel/next.js/discussions/41745#discussioncomment-3973420

Copying for visibility:

> AniminD
> Hi, I think i've seen something in the docs about it, regarding scroll behavior if I'm not mistaking
> "Good to know: Next.js currently wraps each segment in a div element to achieve this behavior. In the future, the div might be removed pending a React API similar to findDOMNode()."


And from Tim's answer:

> It's indeed added for now to handle keeping the right section of the page in view when navigating between pages.
> Unlike in pages the new router has improved scroll and focus handling. It ensures only the changed part is scrolled into view and put focus on.

So yeah, a bit of a boomer for certain HTML semantics. Though: https://caninclude.glitch.me/caninclude?child=main&parent=div

**Reply by volnei** [reply]
There are problems here when using CSS Modules. :(

**Reply by rebelchris** [reply]
Not just semantics, also styling with layouts becomes a pain.

Imagine top layout having:
```js
<main className='flex'>{children}</main>
```

Then child layout something like this:

```js
export default function DashboardLayout({children}: {
    children: React.ReactNode
}) {
    return <>
        <aside className='w-96 bg-cyan-100'>
            Left
        </aside>
        <main>{children}</main>
    </>;
}
```

End result is now:

```html
<main class="flex">
   <div>
      <aside class="w-96 bg-cyan-100">Left</aside>
      <main>
         <div>
            <div>
               <h1>This is my account page</h1>
            </div>
         </div>
      </main>
   </div>
</main>
```

Result is that the flex on the first main doesn't propagate to the aside/main inside because the wrapping div.

**Reply by icyJoseph** [reply]
Yeah even if you were to target class-less divs:only-child divs under main, it can still break in ways you wouldn't want to maintain just because the framework forces you to. Luckily they'll introduce a proper div attribute which can help us target the divs and place display, contents or flow-root on them.


## Edit

As of `next@13.0.1` 

![Screenshot 2022-11-01 at 09 40 43](https://user-images.githubusercontent.com/21013447/199193408-e9001cee-076b-41c5-8f16-af6d6a1c6aec.png)


Also I am able to load my CSS Modules client side normally 😄 no need to pass them as props.

**Reply by AkileshRao** [reply]
Has anyone tried to figure this out yet? This issue still seems to exist and I don't see any suggested workaround in this thread.

**Reply by timneutkens** [reply]
It doesn't. The extra divs were removed in favor of another approach. Maybe you're on an old version of Next.js?

**Reply by vanghoa** [reply]
Hi, I still have that problem in nextjs v13.4.19

**Reply by bryantj1824** [reply]
Currently building a webapp and ran  into the wrapping div issue running version 14.0.4, any workaround?

**Comment by nmarron** [comment]
I can see an extra div on this setup... and it doesn't let the wrapped content to take the right css width (hardcoded in this example)

  "dependencies": {
       "react": "^18",
       "react-dom": "^18",
       "next": "14.1.3"
  },

The Parent:
 ``` 
 <div className={styles["principal-banner-children"]}>
     { children }
</div>
```

The Children:
```
<div style={{width: '100%'}}>
   <h1>Fitness Page</h1>
   <Search />
</div>
```

![Screenshot 2024-03-14 at 5 16 48 PM](https://github.com/vercel/next.js/assets/11777832/500289c3-1ba3-40fb-9410-02bbec3cd178)

  /* Temporary hack since next adds a div */
  .principal-banner-children > div {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }
